services-custom/s3-event-notifications/src/main/java/software/amazon/awssdk/eventnotifications/s3/model/S3EventNotificationRecord.java

```diff
@@ -18,26 +18,57 @@
 import java.time.Instant;
 import java.util.Objects;
 import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.annotations.SdkTestInternalApi;
 import software.amazon.awssdk.utils.ToString;
 
 @SdkPublicApi
 public class S3EventNotificationRecord {
 
-    private final String awsRegion;
-    private final String eventName;
-    private final String eventSource;
-    private final String eventVersion;
-    private final RequestParameters requestParameters;
-    private final ResponseElements responseElements;
-    private final S3 s3;
-    private final UserIdentity userIdentity;
-    private final GlacierEventData glacierEventData;
-    private final LifecycleEventData lifecycleEventData;
-    private final IntelligentTieringEventData intelligentTieringEventData;
-    private final ReplicationEventData replicationEventData;
-    private final Instant eventTime;
-
-    public S3EventNotificationRecord(
+    private String awsRegion;
+    private String eventName;
+    private String eventSource;
+    private String eventVersion;
+    private RequestParameters requestParameters;
+    private ResponseElements responseElements;
+    private S3 s3;
+    private UserIdentity userIdentity;
+    private GlacierEventData glacierEventData;
+    private LifecycleEventData lifecycleEventData;
+    private IntelligentTieringEventData intelligentTieringEventData;
+    private ReplicationEventData replicationEventData;
+    private Instant eventTime;
+
+    public S3EventNotificationRecord() {
+    }
+
+    @SdkTestInternalApi
+    protected S3EventNotificationRecord(
```

Comment:
Should this be package-protected (no modifier) so that people can't get access to the method through extension?